lib/services/iotaRedirector.js

```diff
@@ -162,8 +162,15 @@ function splitByProtocol(req) {
             clonedObject.protocol = req.body[collectionName][i].protocol;
             clonedObject.resource = req.protocolObj[clonedObject.protocol].resource;
 
-            collections[req.body[collectionName][i].protocol] = {};
-            collections[req.body[collectionName][i].protocol][collectionName] = [_.clone(req.body[collectionName][i])];
+            if (collections[req.body[collectionName][i].protocol] &&
+                collections[req.body[collectionName][i].protocol][collectionName]) {
+                collections[req.body[collectionName][i].protocol][collectionName].push(
+                    _.clone(req.body[collectionName][i]));
+            } else {
+                collections[req.body[collectionName][i].protocol] = {};
+                collections[req.body[collectionName][i].protocol][collectionName] =
+                    [_.clone(req.body[collectionName][i])];
+            }
```

Comment:
Fixed in 9fd9344

The solution is not exactly the same, as it may be possible for the collections object to already have data while the array is unexistent.